govuk-notify-spi/src/main/scala/uk/gov/nationalarchives/notifyspi/NotifyEmailSenderProvider.scala

```diff
@@ -0,0 +1,42 @@
+package uk.gov.nationalarchives.notifyspi
+
+import java.util
+
+import org.keycloak.email.EmailSenderProvider
+import org.keycloak.models.UserModel
+import uk.gov.service.notify.{NotificationClient, NotificationClientException}
+
+import scala.jdk.CollectionConverters._
+
+class NotifyEmailSenderProvider(notifyClient: NotificationClient) extends EmailSenderProvider {
+
+  private val templateIdKey = "GOVUK_NOTIFY_TEMPLATE_ID"
+
+  override def send(config: util.Map[String, String],
+                     user: UserModel,
+                     subject: String,
+                     textBody: String,
+                     htmlBody: String): Unit = {
+
```

Comment:
Are we using the html body anywhere. This looks like we'll always be sending a text email out but won't this mean we'll lose any formatting. Or does the notify API sort all that out in the template?

Reply:
The way Notify works is you can send in parameters via the personalisation Map. The is read by the Notify template by referring to the Map key and rendered into the Notify template. Potentially the Map could include htmlBody if we want. At the moment I'm just passing in the standard textBody generated by Keycloak into the Notify template. I've not tried out what would happen if we pass html into the Notify template, and whether it would actually render it.

Reply:
If it works with the text body and it looks right then that's ok. Just wanted to check.

Reply:
Tried sending through html. Notify renders the html as strings in the message, so you get something that looks like this: ".... <p> some text </p>" The tags appear in the message body, so it doesn't look too great. I suspect this is a deliberate to ensure firewalls don't think emails from Notify are spam, as they contain html. Its probably something to look at later and see what Notify / Keycloak can be configured to do.